src/acceptance/assets/app/go_app/internal/app/memory.go

```diff
@@ -1,85 +1,151 @@
 package app
 
 import (
+	"bytes"
 	"container/list"
 	"fmt"
 	"net/http"
+	"os"
+	"runtime"
 	"strconv"
 	"sync"
 	"time"
 
 	"github.com/gin-gonic/gin"
-	"github.com/sirupsen/logrus"
+	"github.com/go-logr/logr"
+	"github.com/prometheus/procfs"
 )
 
-const Megabyte = uint64(1024 * 1024)
+const (
+	_ = 1 << (10 * iota)
+	Kibi
+	Mebi
+)
 
-type MemTestInfo struct {
-	mu        sync.Mutex
+type MemTest struct {
+	mu        sync.RWMutex
 	used      *list.List
 	isRunning bool
 }
 
-var memTest = &MemTestInfo{}
+func MemoryTests(logger logr.Logger, r *gin.RouterGroup, sleep func(duration time.Duration), useMem func(useMb uint64)) *gin.RouterGroup {
```

Comment:
The design of this function looks a bit suprising to me:
 + If you skip one of the input-parameters useCPU or sleep you skip the other one too, even if you actually provided it.
 + If you skip (at least) one of the two parameters mentioned above, the code detects tests that already are running. But otherwise not, when you provide both of them.